### PR TITLE
minor: introduction of validationType in javadocs

### PR DESCRIPTION
**Pull request by gaurabdg**
Contains the remaining modfications of introduction of `validationType`, and changing `int[]` to `String[]` for tokens

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 1]
```diff
@@ -96,6 +96,7 @@
  * Property {@code excludeClassesRegexps} - Specify user-configured regular
  * expressions to ignore classes.
  * Type is {@code java.lang.String[]}.
+ * Validation type is {@code Pattern}.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:99]: Please use fully qualified name for type, I hope we did it for other places.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java [hunk 2]
```diff
@@ -43,7 +43,8 @@
  * <ul>
  * <li>
  * Property {@code limitedTokens} - Specify set of tokens with limited occurrences as descendants.
- * Type is {@code int[]}.
+ * Type is {@code String[]}.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java:46]: For classes in java.lang package we probably can skip fully qualified name, but let's be consistent.
In this PR I see some has fully qualified name for String

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 1 above)
Comment (gaurabdg) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java (hunk 2 above)
Comment (gaurabdg) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java]: Done

**Pull request merged**